applications/MappingApplication/custom_utilities/interface_condition.h

```diff
@@ -93,89 +89,206 @@ namespace Kratos
+                  && mNumPoints == 2) { // I am a linear line condition
+                  is_inside = MapperUtilities::ProjectPointToLine(mpCondition, rGlobalCoords,
                                                                   projection_local_coords,
-                                                                  projection_distance, m_normal);
-          } else if (m_geometry_family == GeometryData::Kratos_Quadrilateral) { // I am a quadrilateral condition
-              is_inside = MapperUtilities::ProjectPointToQuaddrilateral(m_p_condition, global_coords,
-                                                                        projection_local_coords,
-                                                                        projection_distance, m_normal);
-          } else {
-              KRATOS_ERROR << "MappingApplication; InterfaceCondition; \"EvaluateResult\" Unsupported Geometry!" << std::endl;
+                                                                  projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Triangle 
+                        && mNumPoints == 3) { // I am a linear triangular condition
+                  is_inside = MapperUtilities::ProjectPointToTriangle(mpCondition, rGlobalCoords,
+                                                                      projection_local_coords,
+                                                                      projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Quadrilateral
+                        && mNumPoints == 4) { // I am a linear quadrilateral condition
+                  is_inside = MapperUtilities::ProjectPointToQuadrilateral(mpCondition, rGlobalCoords,
+                                                                          projection_local_coords,
+                                                                          projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Tetrahedra ||
+                         mGeometryFamily == GeometryData::Kratos_Prism ||
+                         mGeometryFamily == GeometryData::Kratos_Hexahedra) { // Volume Mapping
+                  is_inside = MapperUtilities::PointLocalCoordinatesInVolume(mpCondition, rGlobalCoords,
+                                                                             projection_local_coords,
+                                                                             projection_distance);
+              } else {
+                  std::cout << "MAPPER WARNING, Unsupported geometry, " 
+                            << "using an approximation" << std::endl;
+                  return false;
+              }
+          } catch (...) {
+              std::cout << "MAPPER WARNING, Functions necessary for projection "
+                        << "are not implemented in the core, " 
+                        << "using an approximation" << std::endl;
+              return false;
           }
```

Comment:
These parts of the code don't throw exceptions, that is true.
However I am using functions of the Geometry such as `GlobalCoordinates()`
If I now use a geometry that does not implement functions that are called by `GlobalCoordinates()`, then functions of the base class (i.e. `Geometry)` are called which throw errors. This is the same behavior as I do for my stuff in case the base class functions are called, see further below.

In such a case the geometry is not suitable for this kind of mapping, because functions are missing in the core. 
I thought it would be a good and especially robust solution if the mapper would still work in such a case.

Another solution would be to use the kratos try catch and print a warning to use the nearestneighbor mapper because of the explained reason.
which might throw errors in case it is not implemented